source/settings/core.rst

```diff
@@ -2868,6 +2868,20 @@ See :ref:`settings` for list of all setting groups.
    As used here, the variable ``%d`` expands to the domain of the local user.
    Other :ref:`mail user variables <variables-mail_user>` can be used as well.
 
+.. dovecot_core:setting:: process_shutdown_filter
+   :values: @string
+
+   .. versionadded:: 2.3.18
+
+   Filter to specify which events stop new connections to the process.  The syntax
+   of the filter is described in :ref:`event_filter_global`.
+
+   For example:
+
+   .. code-block:: none
+
+     process_shutdown_filter = "event=mail_user_session_finished AND vsz > 10M"
```

Comment:
10 MB vsz is likely too low always. I'm not sure yet what is the right value, but maybe use `rss > 10M` for now, which should be at least closer.